server/commands/src/impls.rs

```diff
@@ -573,3 +578,242 @@ pub fn seed(ctx: &mut CommandCtx) -> anyhow::Result<()> {
+    }
+
+    let ent = Read::<(SocketAddr, Entity)>::query()
+        .iter(ctx.world.inner())
+        .find(|x| x.0.ip() == ip)
+        .map(|x| (*x).1);
+
+    if let Some(ent) = ent {
+        ctx.game.disconnect_and_log(
+            ent,
+            &mut ctx.world,
+            &Text::from(reason),
+            "Banned by an operator.",
+        );
+    }
+
+    Ok(None)
+}
+
+pub fn ban_players(
+    ctx: &mut CommandCtx,
+    targets: EntitySelector,
+    reason: String,
+    by_ip: bool,
+) -> anyhow::Result<Option<String>> {
+    if targets.entities.is_empty() {
+        return Err(BanError::NoTargets.into());
+    }
+
+    for entity in &targets.entities {
+        if ctx.world.try_get::<Player>(*entity).is_none() {
+            return Err(BanError::NotPlayer.into());
+        }
+    }
+
+    for entity in &targets.entities {
+        {
+            let bi_lock = ctx.game.resources.get::<WrappedBanInfo>();
+            let mut ban_info = bi_lock.write().unwrap();
+
+            if by_ip {
+                let ip = ctx.world.try_get::<SocketAddr>(*entity).unwrap();
+
+                ban_info.ip_bans.insert(
+                    ip.ip(),
+                    Ban {
+                        reason: reason.clone(),
+                        expires_after: None,
+                    },
+                );
+            } else {
+                let uuid = ctx.world.try_get::<Uuid>(*entity).unwrap();
+
+                ban_info.uuid_bans.insert(
+                    uuid.to_hyphenated_ref().to_string(),
+                    Ban {
+                        reason: reason.clone(),
+                        expires_after: None,
+                    },
+                );
+            }
+        }
+
+        let name = ctx.world.try_get::<Name>(*entity).unwrap().0.clone();
+        if let Some(mut sender_message_receiver) =
+            ctx.world.try_get_mut::<MessageReceiver>(ctx.sender)
+        {
+            let ban_confirm = Text::from(TextValue::translate_with(
+                "commands.ban.success",
+                vec![Text::from(name), Text::from(reason.clone())],
+            ));
+            sender_message_receiver.send(ban_confirm);
+        }
+
+        ctx.game.disconnect_and_log(
+            *entity,
+            &mut ctx.world,
+            &Text::from(reason.clone()),
+            "Banned by an operator.",
+        );
+    }
+
+    Ok(None)
+}
+
+#[derive(Debug, Error)]
+pub enum PardonError {
+    #[error("Couldn't find that players UUID, Have they changed name?")]
+    NotPlayer,
+}
+
+#[command(usage = "pardon <name>")]
+pub fn pardon(ctx: &mut CommandCtx, name: TextArgument) -> anyhow::Result<()> {
+    // Get UUID from name
+    let online_mode = ctx.game.shared.config.server.online_mode;
+    let uuid = if online_mode {
+        Runtime::new()
+            .unwrap()
+            .block_on(name_to_uuid_online(&name.0))
+    } else {
+        Some(name_to_uuid_offline(&name.0))
+    };
+
+    if uuid.is_none() {
```

Comment:
This could be more idiomatic with a match:

```rust
let uuid = match uuid {
    Some(uuid) => uuid,
    None => return Err(PardonError::NotPlayer.into()),
}
```